recipes/influxdb-cpp/all/conanfile.py

```diff
@@ -0,0 +1,39 @@
+import os
+
+from conans import ConanFile, tools
+import glob
+
+
+class InfluxDBCppConan(ConanFile):
+    name = "influxdb-cpp"
+    license = "MIT"
+    homepage = "https://github.com/orca-zhang/influxdb-cpp"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "C++ client for InfluxDB."
+    topics = ("single-header-lib", "influxdb")
+    settings = "os"
+    exports_sources = ["patches/**"]
+    no_copy_source = True
+
+    _source_subfolder = "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("influxdb-cpp-*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+        patches = self.conan_data["patches"][self.version]
+        for patch in patches:
+            tools.patch(**patch)
```

Comment:
patches need to be applied in the build method!

you'll need to remove `no_copy_source` for the build() to be called

Reply:
The include guards I added are to prevent this kind of redefinition error:

```
C:\Users\U000491\.conan\data\influxdb-cpp\cci.20200712\_\_\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\inf
luxdb.hpp(16,26): error C2011: 'iovec': 'struct' type redefinition [C:\projects\conan-center-index\recipes\influxdb-cpp
\all\test_package\build\ecbfb49a4e2ea8618319be5f9e02c8e0621391fb\example.vcxproj]
```

The removal of:

```c++
#define NOMINMAX
```
is to prevent also some issues with compilers complaining about redefinition of `NOMINMAX` and, as the library depends on it being defined on a global level, I think this would fit better in a `cpp_info.defines` to enforce it being defined transitively by consuming projects.

The removal of:

```c++
#pragma comment(lib, "ws2_32")
```
is to turn the library more suitable to Conan by moving such a linkage specification out of the source into a more appropriate place in `cpp_info.system_libs`.